sklearn/cluster/tests/test_k_medoids.py

```diff
@@ -0,0 +1,139 @@
+"""Testing for K-Medoids"""
+import numpy as np
+from scipy.sparse import csc_matrix
+
+from sklearn.cluster import KMedoids, KMeans
+from sklearn.datasets import load_iris
+from sklearn.exceptions import NotFittedError
+from sklearn.metrics.pairwise import PAIRWISE_DISTANCE_FUNCTIONS
+from sklearn.metrics.pairwise import euclidean_distances
+from sklearn.utils.testing import assert_array_equal
+from sklearn.utils.testing import assert_equal, assert_raises, \
+    assert_raise_message
+from sklearn.utils.testing import assert_greater
+from sklearn.utils.testing import assert_allclose
+
+rng = np.random.RandomState(0)
+X = rng.rand(100, 5)
+
+
+def test_kmedoids_input_validation_and_fit_check():
+    # Invalid parameters
+    assert_raise_message(ValueError, "n_clusters should be a nonnegative "
+                                     "integer. 0 was given",
+                         KMedoids(n_clusters=0).fit, X)
+
+    assert_raise_message(ValueError, "n_clusters should be a nonnegative "
+                                     "integer. None was given",
+                         KMedoids(n_clusters=None).fit, X)
+
+    assert_raise_message(ValueError, "init needs to be one of the following: "
+                                     "['random', 'heuristic']",
+                         KMedoids(init=None).fit, X)
+
+    # Trying to fit 3 samples to 8 clusters
+    Xsmall = rng.rand(5, 2)
+    assert_raise_message(ValueError, "The number of medoids 8 must be less "
+                                     "than the number of samples 5.",
+                         KMedoids(n_clusters=8).fit, Xsmall)
+
+    # Test if NotFittedError is raised appropriately
```

Comment:
Common tests  should do this

Reply:
by should you mean they do and this can be safely removed? :)